From a7be6d8cdc16cf75394a0950675b638c1003e575 Mon Sep 17 00:00:00 2001
Message-ID: <a7be6d8cdc16cf75394a0950675b638c1003e575.1705743628.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1705743627.git.sharlatanus@gmail.com>
References: <cover.1705743627.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 16 Jan 2024 23:50:02 +0000
Subject: [PATCH 21/38] gnu: go-github-com-btcsuite-btcd-btcec: Move to (gnu
 packages golang-crypto).

* gnu/packages/golang.scm (go-github-com-btcsuite-btcd-btcec): Move from
here ...
* gnu/packages/golang-crypto.scm: ... to here.

Change-Id: Iea48f1ba1edcf9daf567afc3ed1133f0895a0e99
---
 gnu/packages/golang-crypto.scm | 38 ++++++++++++++++++++++++++++++++++
 gnu/packages/golang.scm        | 38 ----------------------------------
 2 files changed, 38 insertions(+), 38 deletions(-)

diff --git a/gnu/packages/golang-crypto.scm b/gnu/packages/golang-crypto.scm
index 417dbc34f6..91def02194 100644
--- a/gnu/packages/golang-crypto.scm
+++ b/gnu/packages/golang-crypto.scm
@@ -1,5 +1,6 @@
 ;;; GNU Guix --- Functional package management for GNU
 ;;; Copyright © 2018 Pierre Neidhardt <mail@ambrevar.xyz>
+;;; Copyright © 2019 Vagrant Cascadian <vagrant@debian.org>
 ;;; Copyright © 2019, 2020 Leo Famulari <leo@famulari.name>
 ;;; Copyright © 2021 Raghav Gururajan <rg@raghavgururajan.name>
 ;;; Copyright © 2022 (unmatched-parenthesis <paren@disroot.org>
@@ -124,6 +125,43 @@ (define-public go-github-com-aperturerobotics-jacobsa-crypto
 cryptographic standards that are not included in the Go standard library.")
       (license license:asl2.0))))
 
+(define-public go-github-com-btcsuite-btcd-btcec
+  (let ((commit "67e573d211ace594f1366b4ce9d39726c4b19bd0")
+        (revision "0"))
+    (package
+      (name "go-github-com-btcsuite-btcd-btcec")
+      (version (git-version "0.12.0-beta" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/btcsuite/btcd")
+               (commit commit)))
+         (file-name (git-file-name name version))
+         (sha256
+          (base32 "04s92gsy71w1jirlr5lkk9y6r5cparbas7nmf6ywbp7kq7fn8ajn"))))
+      (build-system go-build-system)
+      (arguments
+       '(#:unpack-path "github.com/btcsuite/btcd"
+         #:import-path "github.com/btcsuite/btcd/btcec"))
+      (native-inputs
+       (list go-github-com-davecgh-go-spew))
+      (home-page "https://github.com/btcsuite/btcd")
+      (synopsis "Elliptic curve cryptography to work with Bitcoin")
+      (description "Package @command{btcec} implements elliptic curve
+cryptography needed for working with Bitcoin (secp256k1 only for now).  It is
+designed so that it may be used with the standard crypto/ecdsa packages
+provided with Go.  A comprehensive suite of test is provided to ensure proper
+functionality.  Package @command{btcec} was originally based on work from
+ThePiachu which is licensed under the same terms as Go, but it has
+significantly diverged since then.  The @command{btcsuite} developers original
+is licensed under the liberal ISC license.
+
+Although this package was primarily written for btcd, it has intentionally
+been designed so it can be used as a standalone package for any projects
+needing to use secp256k1 elliptic curve cryptography.")
+      (license license:isc))))
+
 (define-public go-github-com-emersion-go-pgpmail
   (package
     (name "go-github-com-emersion-go-pgpmail")
diff --git a/gnu/packages/golang.scm b/gnu/packages/golang.scm
index 4de3d321ed..562d1adcfa 100644
--- a/gnu/packages/golang.scm
+++ b/gnu/packages/golang.scm
@@ -5596,44 +5596,6 @@ (define-public go-github-com-btcsuite-btclog
 implementing the same interface.")
       (license license:isc))))
 
-(define-public go-github-com-btcsuite-btcd-btcec
-  (let ((commit "67e573d211ace594f1366b4ce9d39726c4b19bd0")
-        (revision "0"))
-    (package
-      (name "go-github-com-btcsuite-btcd-btcec")
-      (version (git-version "0.12.0-beta" revision commit))
-      (source
-       (origin
-         (method git-fetch)
-         (uri (git-reference
-               (url "https://github.com/btcsuite/btcd")
-               (commit commit)))
-         (file-name (git-file-name name version))
-         (sha256
-          (base32
-           "04s92gsy71w1jirlr5lkk9y6r5cparbas7nmf6ywbp7kq7fn8ajn"))))
-      (build-system go-build-system)
-      (arguments
-       '(#:unpack-path "github.com/btcsuite/btcd"
-         #:import-path "github.com/btcsuite/btcd/btcec"))
-      (native-inputs
-       (list go-github-com-davecgh-go-spew))
-      (home-page "https://github.com/btcsuite/btcd")
-      (synopsis "Elliptic curve cryptography to work with Bitcoin")
-      (description "Package @command{btcec} implements elliptic curve
-cryptography needed for working with Bitcoin (secp256k1 only for now).  It is
-designed so that it may be used with the standard crypto/ecdsa packages
-provided with Go.  A comprehensive suite of test is provided to ensure proper
-functionality.  Package @command{btcec} was originally based on work from
-ThePiachu which is licensed under the same terms as Go, but it has
-significantly diverged since then.  The @command{btcsuite} developers original
-is licensed under the liberal ISC license.
-
-Although this package was primarily written for btcd, it has intentionally
-been designed so it can be used as a standalone package for any projects
-needing to use secp256k1 elliptic curve cryptography.")
-      (license license:isc))))
-
 (define-public go-github-com-minio-sha256-simd
   (package
     (name "go-github-com-minio-sha256-simd")
-- 
2.41.0

